### ISSUE TITLE
Building 4.0.14 in x64 Release configuration creates an unusable executable?

### ISSUE THREAD

**Issue by assaf-arbely** [tporadowski/redis#49]
Hi,
First, @tporadowski - Thanks for your wonderful work!

Trying to build the 4.0.14 in x64 Release configuration, using MSVS 2019 community, with Win 10 SDK and latest WIX. Built all components and created an MSI. Tried to build on both, Win 7 and Win 10 hosts.

When installing the MSI on Win 2016 server (local admin, elevated session), service can't start.

When repeating the same build in Debug configuration - everything works smoothly!

Any idea? I most probably miss something...

Thanks in advance, Assaf

**Comment by tporadowski** [comment]
Does `redis-server.exe` start from command-line on that server? Is there anything logged in Event Log related to that unsuccessful service start?

**Comment by assaf-arbely** [comment]
Yes, redis-server.exe can start from the command line on that server, as long as no parameter is provided. No meaningful error in windows Event Log.

**Comment by tporadowski** [comment]
I suppose the [4.0.14 version built by me](https://github.com/tporadowski/redis/releases/tag/v4.0.14) is no different. Currently I have no test environment with Windows Server 2016, so cannot really test it.

Btw. what happens if you provide any parameters when running from command-line?

**Comment by assaf-arbely** [comment]
Thanks for a prompt reply. You are much faster than me (: 
Here are the answers to your questions, together with some more observations.

**a.** When starting from an elevated command-line with parameters (redis-server.exe --help), we get an error message in a window titled with: "Redis for Windows, based on OpenTech port has stopped working"

**b.** During the install of Release, the installer tries to start the service and fails with an error:
"Service 'Redis' (Redis) failed to start. Verify that you have sufficient privileges to start system services."

**c.** (As opposed to what I said before) In the Event Log, after submitting the command line in item a, we see two errors:
1) syslog-ident = redis
Invalid argument during startup: unknown argument:

2) Faulting application name: redis-server.exe, version: 4.0.14.0, time stamp: 0x5e1b521e
Faulting module name: redis-server.exe, version: 4.0.14.0, time stamp: 0x5e1b521e
Exception code: 0xc0000094
Fault offset: 0x0000000000008ff8
Faulting process id: 0x764
Faulting application start time: 0x01d5caf1fa95955d
Faulting application path: C:\Program Files\Redis\redis-server.exe
Faulting module path: C:\Program Files\Redis\redis-server.exe
Report Id: 441efd47-8454-44a4-a9a9-67372b6bc645
Faulting package full name:
Faulting package-relative application ID:

**Comment by tporadowski** [comment]
This seems to be related to user permissions, especially given that message related to _"Verify that you have sufficient privileges to start system services."._ I have just tested 4.0.14 MSI installer on a Windows Server 2016 Standard Edition (running in VirtualBox) where I installed it from administrator account - and everything works fine. `systeminfo` there shows:
```
OS Name:                   Microsoft Windows Server 2016 Standard Evaluation
OS Version:                10.0.14393 N/A Build 14393
```
`whoami` output: `win-57jgtv3sgft\administrator`

**Comment by tporadowski** [comment]
@assaf-arbely I have just created [4.0.14.1 release](https://github.com/tporadowski/redis/releases/tag/v4.0.14.1) that contains fixes for #46 and #47 and while testing it I noticed the same behavior you were seeing which was related to issue #47. Please try that version and let me know if it works for you as well (or if you want to build it yourself take code from tag `v4.0.14.1`).

**Comment by assaf-arbely** [comment]
@tporadowski You are right, it actually fixed the issue.
Thank you so much for taking care of that.
Would you like me to close this issue?
